lib/code_corps/messages/emails.ex

```diff
@@ -0,0 +1,93 @@
+defmodule CodeCorps.Messages.Emails do
+  @moduledoc """
+  Handles email notifications used within the Messages context
+  """
+  alias CodeCorps.{
+    Conversation,
+    ConversationPart,
+    Emails,
+    Mailer,
+    Message,
+    Repo
+  }
+
+  @message_preloads [:project, [conversations: :user]]
+
+  @doc ~S"""
+  Notifies all relevant users of a new `CodeCorps.Message` being created.
+
+  We match against admin initiated messages only, because it's unclear who
+  message.author is for user-initiated messages, or who the email recipient
+  would be, assuming the message.author is the user who initiated the message.
+  """
+  @spec notify_of_new_message(Message.t) :: :ok
+  def notify_of_new_message(%Message{initiated_by: "admin"} = message) do
+    message = message |> Repo.preload(@message_preloads)
+
+    message
+    |> Map.get(:conversations)
+    |> Enum.map(&Emails.MessageForUserEmail.create(message, &1))
+    |> Enum.each(&Mailer.deliver_now/1)
+  end
+
+  @part_preloads [
+    conversation: [
+      [conversation_parts: :author],
+      [message: [:author, [project: :organization]]],
+      :user
+    ]
+  ]
+
+  @doc ~S"""
+  Notifies via email all relevant users when a new `CodeCorps.ConvrsationPart`
+  has been added to a `CodeCorps.Conversation`.
+  """
+  @spec notify_of_new_reply(ConvrsationPart.t) :: :ok
+  def notify_of_new_reply(%ConversationPart{} = part) do
+    part = part |> Repo.preload(@part_preloads)
+
+    part |> maybe_send_message_for_user_email()
+    part |> send_message_for_project_emails()
+  end
+
+  # we match against admin initiated only, because it's unclear who
+  # message.author is for user-initiated messages, or who the email recipient
+  # would be, assuming the message.author is the user who initiated the message
+  @spec send_message_for_project_emails(ConversationPart.t) :: :ok
```

Comment:
As the comment says

- who gets notified of an email sent by a user to a project? I know we don't have it yet, but still
- who gets assigned as the message.author. Having it be identical to conversation.user seems redundant/confusing. Should we allow it blank in that case?

Reply:
I think our conversation earlier got it right. Can you comment here what we said?

Reply:
IIRC, the idea was

When a user messages a project, all project admins and owners receive an email.

When a project admin/owner replies to the conversation, from that point on, only participating users receive emails. That means whoever send it initially, whoever replied, and whoever decided to join in at a later time.